app/src/include/firebase/internal/future_impl.h

```diff
@@ -189,11 +189,14 @@ inline FutureBase::CompletionCallbackHandle Future<ResultType>::AddOnCompletion(
 
 #endif  // defined(INTERNAL_EXPERIMENTAL)
 
-inline FutureBase::FutureBase() : api_(NULL), handle_(0) {}  // NOLINT
+inline FutureBase::FutureBase()
+    : mutex_(Mutex::Mode::kModeNonRecursive),
+      api_(NULL),
+      handle_(0) {}  // NOLINT
 
 inline FutureBase::FutureBase(detail::FutureApiInterface* api,
                               const FutureHandle& handle)
-    : api_(api), handle_(handle) {
+    : mutex_(Mutex::Mode::kModeNonRecursive), api_(api), handle_(handle) {
```

Comment:
Is this in here just because std::mutex happens to be nonrecursive, so we know it works?

Reply:
Basically, yes. It will make it less "scary" to convert to std::mutex some day in the future since we know it's works in non-recursive mode.

Reply:
Should I make a subclass of `Mutex` called `NonRecursiveMutex` which simply specifies `Mutex::Mode::kModeNonRecursive` to the superclass' constructor? That would avoid having to explicitly initialize the `Mutex` member variable and could encourage the use of non-recursive mutexes, making it easier to port to `std::mutex` some day.

Reply:
Nah, I think it's fine as-is. Was just curious.